src/ZendCodingStandard/ruleset.xml

```diff
@@ -201,232 +70,823 @@
         <!-- Doesn't respect inheritance -->
-        <exclude name="PEAR.Commenting.FunctionComment.ScalarTypeHintMissing"/>
+        <exclude name="Squiz.Commenting.FunctionComment.ScalarTypeHintMissing"/>
         <!-- Throws comments can start without a capital and doesn't need a full stop -->
         <exclude name="PEAR.Commenting.FunctionComment.ThrowsNoFullStop"/>
         <exclude name="PEAR.Commenting.FunctionComment.ThrowsNotCapital"/>
         <!-- Doesn't work with self as typehint -->
-        <exclude name="PEAR.Commenting.FunctionComment.TypeHintMissing"/>
-    </rule>
-    <!-- Forbid `AND` and `OR`, require `&&` and `||` -->
-    <rule ref="Squiz.Operators.ValidLogicalOperators"/>
-    <!-- Forbid `global` -->
-    <rule ref="Squiz.PHP.GlobalKeyword"/>
-    <!-- Require PHP function calls in lowercase -->
-    <rule ref="Squiz.PHP.LowercasePHPFunctions"/>
-    <!-- Forbid dead code -->
-    <rule ref="Squiz.PHP.NonExecutableCode"/>
-    <!-- Forbid `$this` inside static function -->
-    <rule ref="Squiz.Scope.StaticThisUsage"/>
-    <!-- Force whitespace before and after concatenation -->
-    <rule ref="Squiz.Strings.ConcatenationSpacing">
-        <properties>
-            <property name="spacing" value="1"/>
-            <property name="ignoreNewlines" value="true"/>
-        </properties>
+        <exclude name="Squiz.Commenting.FunctionComment.TypeHintMissing"/>
     </rule>
-    <!-- Forbid strings in `"` unless necessary -->
-    <rule ref="Squiz.Strings.DoubleQuoteUsage"/>
-    <rule ref="Squiz.Strings.DoubleQuoteUsage.ContainsVar">
-        <message>Variable "%s" not allowed in double quoted string; use sprintf() or concatenation instead</message>
+
+    <!-- The asterisks in a DocBlock should align, and there should be one
+         space between the asterisk and tag. -->
+    <rule ref="Squiz.Commenting.DocCommentAlignment">
+        <exclude name="Squiz.Commenting.DocCommentAlignment.SpaceAfterStar"/>
     </rule>
-    <!-- Forbid braces around string in `echo` -->
-    <rule ref="Squiz.Strings.EchoedStrings"/>
-    <!-- Forbid spaces in type casts -->
-    <rule ref="Squiz.WhiteSpace.CastSpacing"/>
-    <!-- Require space after language constructs -->
-    <rule ref="Squiz.WhiteSpace.LanguageConstructSpacing"/>
-    <!-- Require space around logical operators -->
-    <rule ref="Squiz.WhiteSpace.LogicalOperatorSpacing"/>
-    <!-- Forbid spaces around `->` operator -->
-    <rule ref="Squiz.WhiteSpace.ObjectOperatorSpacing">
+
+    <!-- DocBlocks MUST follow this specific order of annotations with empty
+         newline between specific groups:
+
+         /**
+          * <Summary>
+          *
+          * <Description>
+          *
+          * @internal
+          * @deprecated
+          *
+          * @link
+          * @see
+          * @uses
+          *
+          * @param
+          *
+          * @return
+          *
+          * @throws
```

Comment:
We never discussed this.

I've traditionally always grouped these last three in this same order, but without whitespace between the elements. Why are we adding whitespace? Can you point to information on this decision somewhere?

Reply:
It's not discussed yet but I stumbled upon this rule somewhere and included it. I'll keep the order and remove the whitespace.

Reply:
Cool - thanks. Looks like you can create an annotation group similar to the way things like `@internal` and whatnot are done.

Reply:
Yes exactly that. You can force the order and whitespacing you want. If it's not the order, cs-fix will reorder it. Any suggestions for the ordering?

Reply:
Exactly as it is: params, return value, throw statements.